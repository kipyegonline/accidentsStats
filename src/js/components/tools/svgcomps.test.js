import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { SVG, Group, Rect, Text, Circle, Path } from "./SvgComp";

import { data } from "../../__tests__/table.test";

Enzyme.configure({ adapter: new Adapter() });

//svg comp
describe("<SVG/> component", () => {
  const wrapper = shallow(
    <SVG height={100} bg="#111" width={200} classlist="svg" />
  );

  it("exists", () => expect(wrapper.hasClass("svg")).toBeTruthy());
  it("has props", () =>
    expect(wrapper.props()).toEqual({
      height: 100,
      width: 200,
      strokeWidth: 1,
      style: { background: "#111" },
      className: "svg",
      stroke: "none",
      children: "",
    }));
});

//Group
describe("<Group/> component", () => {
  const wrapper = shallow(
    <Group x={100} y={200} gh={480} gw={400} fill={"grey"} classlist="group" />
  );

  it("renders", () => expect(wrapper.find("g")).toHaveLength(1));
  it("Has props", () =>
    expect(wrapper.props()).toEqual({
      children: "",
      transform: "translate(100,200)",
      height: 480,
      width: 400,
      fill: "grey",
      className: "group",
      shapeRendering: "auto",
    }));
});
//Rect
describe("<Rect/> component", () => {
  const hc = jest.fn(),
    hh = jest.fn(),
    hl = jest.fn();
  const wrapper = shallow(
    <Rect
      x={50}
      y={100}
      width={20}
      height={40}
      fill="grey"
      opacity={0.7}
      style={{ fontSize: "1rem", fontFamily: "raleway" }}
      classlist="rect"
      stroke="none"
      strokew={1}
      hc={hc}
      hl={hl}
      hh={hh}
    />
  );

  describe("renders the UI", () => {
    it("renders a rect componet", () =>
      expect(wrapper.find(".rect")).toHaveLength(1));
  });
  it("has props", () =>
    expect(wrapper.props()).toEqual({
      x: 50,
      onClick: hc,
      onMouseEnter: hh,
      onMouseOver: hh,
      onMouseLeave: hl,
      y: 100,
      width: 20,
      height: 40,
      fill: "grey",
      opacity: 0.7,
      style: { fontSize: "1rem", fontFamily: "raleway" },
      className: "rect",
      shapeRendering: "crispEdges",
      stroke: "none",
      strokeWidth: 1,
    }));
  test(" mouse over event", () => {
    wrapper.simulate("mouseover");
    expect(hh).toHaveBeenCalled();
    expect(hh).toHaveBeenCalledTimes(1);
  });
  test(" mouse leave event", () => {
    wrapper.simulate("mouseleave");
    expect(hl).toHaveBeenCalledTimes(1);
    expect(hl).toHaveBeenCalled();
  });
  test(" click event", () => {
    wrapper.simulate("click");
    expect(hl).toHaveBeenCalledTimes(1);
    expect(hl).toHaveBeenCalled();
  });
});

//TEXT component
describe("<TEXT/> component", () => {
  const wrapper = shallow(
    <Text
      x={30}
      y={60}
      transformer={0}
      fw="normal"
      fz="2rem"
      TA="middle"
      fill="teal"
      classlist="text"
      style={{ fontStyle: "italic" }}
    >
      Say Ahh
    </Text>
  );

  describe("Renders UI", () => {
    it("exists", () => expect(wrapper.find(".text").exists()).toBeTruthy());
    it("has some children", () => expect(wrapper.text()).toBe("Say Ahh"));

    it("has the damn props", () =>
      expect(wrapper.props()).toEqual({
        x: 30,
        y: 60,
        pointerEvents: "none",
        transform: `translate(0)`,
        children: "Say Ahh",
        fontWeight: "normal",
        fill: "teal",
        fontSize: "2rem",
        textAnchor: "middle",
        className: "text",
        style: { fontStyle: "italic" },
      }));
  });
});

describe("<Circle/>", () => {
  const hh = jest.fn(),
    hl = jest.fn();
  const wrapper = shallow(
    <Circle
      cx={100}
      cy={70}
      r={40}
      fill="skyblue"
      opacity={1}
      classlist="circle"
      hh={hh}
      hl={hl}
    >
      <Text x={30} y={70}>
        Welcome to the blue story
      </Text>
    </Circle>
  );
  test("That it renders", () =>
    expect(wrapper.find(".circle").exists()).toBeTruthy());
  test.skip("The circle props", () =>
    expect(wrapper.props()).toEqual({
      cx: 100,
      cy: 70,
      r: 40,
      fill: "skyblue",
      opacity: 1,
      className: "circle",
      onMouseEnter: hh,
      onMouseLeave: hl,
      onMouseOver: hh,
      stroke: "none",
      strokeWidth: 2,
      strokeDashArray: 0,
      shapeRendering: "auto",
      children: "<text>Welcome to the blue story</text>", //plus everything up there
    }));
  test("Mouse events", () => {
    wrapper.simulate("mouseover");
    expect(hh).toHaveBeenCalled();
    expect(hh).toHaveBeenCalledTimes(1);
    wrapper.simulate("mouseleave");
    expect(hl).toHaveBeenCalled();
    expect(hl).toHaveBeenCalledTimes(1);
  });
});

describe("<Path/>", () => {
  const hh = jest.fn(),
    hl = jest.fn();
  const wrapper = shallow(
    <Path
      d="SOME FUNNY PATH STRING TO BE GENERATED BY D# LINE METHOD"
      stroke="red"
      strokew={3}
      fill="#fff"
      hh={hh}
      hl={hl}
      classlist="path"
      opacity={1}
    />
  );
  it("exists", () => expect(wrapper.find(".path").exists()).toBeTruthy());
  it("renders the damn props....", () =>
    expect(wrapper.props()).toEqual({
      d: "SOME FUNNY PATH STRING TO BE GENERATED BY D# LINE METHOD",
      stroke: "red",
      strokeWidth: 3,
      fill: "#fff",
      onMouseLeave: hl,
      onMouseOver: hh,
      shapeRendering: "auto",
      className: "path",
      opacity: 1,
    }));
});
