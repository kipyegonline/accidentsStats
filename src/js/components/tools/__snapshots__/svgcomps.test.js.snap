// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableBody/> renders 1`] = `"<td class=\\"drivers-fatalties-2020\\">345</td>"`;

exports[`<TableStats/> matches snapshot 1`] = `
<table
  className="table table-hover table-bordered border-outline-primary  position-relative my-3 table"
>
  <thead>
    <tr>
      <th
        className="font-weight-bold"
      >
        Victim class
      </th>
      <th
        className="font-weight-bold "
      >
        Summary
      </th>
      <TableList
        TableBody={[Function]}
        a="victimClass"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        sendValue={[Function]}
      >
        <TableHeader
          classlist="drivers"
          key="1"
          name="drivers"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            }
          }
        >
          <th
            className="font-weight-bold"
          >
            Drivers
          </th>
        </TableHeader>
        <TableHeader
          classlist="pillion_passengers"
          key="2"
          name="pillion_passengers"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            }
          }
        >
          <th
            className="font-weight-bold"
          >
            Pillion passengers
          </th>
        </TableHeader>
        <TableHeader
          classlist="pedestrians"
          key="3"
          name="pedestrians"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            }
          }
        >
          <th
            className="font-weight-bold"
          >
            Pedestrians
          </th>
        </TableHeader>
        <TableHeader
          classlist="passengers"
          key="4"
          name="passengers"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            }
          }
        >
          <th
            className="font-weight-bold"
          >
            Passengers
          </th>
        </TableHeader>
        <TableHeader
          classlist="motor_cyclists"
          key="5"
          name="motor_cyclists"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            }
          }
        >
          <th
            className="font-weight-bold"
          >
            Motor cyclists
          </th>
        </TableHeader>
      </TableList>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Fatalties
      </td>
      <td
        className="font-weight-bold"
      >
        1,460
      </td>
      <TableList
        TableBody={[Function]}
        a="fatalties"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        sendValue={[Function]}
      >
        <TableBody
          classlist="drivers"
          key="1"
          name={345}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            }
          }
        >
          <td
            className="drivers-fatalties-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            345
          </td>
        </TableBody>
        <TableBody
          classlist="pillion_passengers"
          key="2"
          name={245}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            }
          }
        >
          <td
            className="pillion_passengers-fatalties-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            245
          </td>
        </TableBody>
        <TableBody
          classlist="pedestrians"
          key="3"
          name={445}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            }
          }
        >
          <td
            className="pedestrians-fatalties-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            445
          </td>
        </TableBody>
        <TableBody
          classlist="passengers"
          key="4"
          name="221"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            }
          }
        >
          <td
            className="passengers-fatalties-2019"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            221
          </td>
        </TableBody>
        <TableBody
          classlist="motor_cyclists"
          key="5"
          name="204"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            }
          }
        >
          <td
            className="motor_cyclists-fatalties-2019"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            204
          </td>
        </TableBody>
      </TableList>
    </tr>
    <tr>
      <td>
        Seriously Injured
      </td>
      <td
        className="font-weight-bold"
      >
        2,733
      </td>
      <TableList
        TableBody={[Function]}
        a="seriouslyInjured"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        sendValue={[Function]}
      >
        <TableBody
          classlist="drivers"
          key="1"
          name={654}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            }
          }
        >
          <td
            className="drivers-seriouslyInjured-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            654
          </td>
        </TableBody>
        <TableBody
          classlist="pillion_passengers"
          key="2"
          name={554}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            }
          }
        >
          <td
            className="pillion_passengers-seriouslyInjured-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            554
          </td>
        </TableBody>
        <TableBody
          classlist="pedestrians"
          key="3"
          name={454}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            }
          }
        >
          <td
            className="pedestrians-seriouslyInjured-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            454
          </td>
        </TableBody>
        <TableBody
          classlist="passengers"
          key="4"
          name="715"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            }
          }
        >
          <td
            className="passengers-seriouslyInjured-2019"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            715
          </td>
        </TableBody>
        <TableBody
          classlist="motor_cyclists"
          key="5"
          name="356"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            }
          }
        >
          <td
            className="motor_cyclists-seriouslyInjured-2019"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            356
          </td>
        </TableBody>
      </TableList>
    </tr>
    <tr>
      <td>
        Slightly injured
      </td>
      <td
        className="font-weight-bold"
      >
        2,392
      </td>
      <TableList
        TableBody={[Function]}
        a="slightlyInjured"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        sendValue={[Function]}
      >
        <TableBody
          classlist="drivers"
          key="1"
          name={435}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            }
          }
        >
          <td
            className="drivers-slightlyInjured-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            435
          </td>
        </TableBody>
        <TableBody
          classlist="pillion_passengers"
          key="2"
          name={405}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            }
          }
        >
          <td
            className="pillion_passengers-slightlyInjured-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            405
          </td>
        </TableBody>
        <TableBody
          classlist="pedestrians"
          key="3"
          name={535}
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            }
          }
        >
          <td
            className="pedestrians-slightlyInjured-2020"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            535
          </td>
        </TableBody>
        <TableBody
          classlist="passengers"
          key="4"
          name="919"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            }
          }
        >
          <td
            className="passengers-slightlyInjured-2019"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            919
          </td>
        </TableBody>
        <TableBody
          classlist="motor_cyclists"
          key="5"
          name="98"
          sendValue={[Function]}
          yoda={
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            }
          }
        >
          <td
            className="motor_cyclists-slightlyInjured-2019"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
          >
            98
          </td>
        </TableBody>
      </TableList>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <td
        className="font-weight-bold"
      >
        Total
      </td>
      <td
        className="purple darken-1 text-white text-lg"
      >
        6,585
      </td>
      <TableFoot
        a="drivers"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        key="0"
      >
        <td
          className="font-weight-bold summary_drivers_2020"
          data-hover="summary_drivers_2020"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1,434
        </td>
      </TableFoot>
      <TableFoot
        a="pillion_passengers"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        key="1"
      >
        <td
          className="font-weight-bold summary_pillion_passengers_2020"
          data-hover="summary_pillion_passengers_2020"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1,204
        </td>
      </TableFoot>
      <TableFoot
        a="pedestrians"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        key="2"
      >
        <td
          className="font-weight-bold summary_pedestrians_2020"
          data-hover="summary_pedestrians_2020"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1,434
        </td>
      </TableFoot>
      <TableFoot
        a="passengers"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        key="3"
      >
        <td
          className="font-weight-bold summary_passengers_2020"
          data-hover="summary_passengers_2020"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          1,855
        </td>
      </TableFoot>
      <TableFoot
        a="motor_cyclists"
        data={
          Array [
            Object {
              "addedon": "2020-02-14",
              "fatalties": 345,
              "fill": "green",
              "id": 1,
              "seriouslyInjured": 654,
              "slightlyInjured": 435,
              "victimClass": "drivers",
            },
            Object {
              "addedon": "2020-03-14",
              "fatalties": 245,
              "fill": "yellow",
              "id": 2,
              "seriouslyInjured": 554,
              "slightlyInjured": 405,
              "victimClass": "pillion_passengers",
            },
            Object {
              "addedon": "2020-05-14",
              "fatalties": 445,
              "fill": "red",
              "id": 3,
              "seriouslyInjured": 454,
              "slightlyInjured": 535,
              "victimClass": "pedestrians",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "221",
              "fill": "orange",
              "id": 4,
              "seriouslyInjured": "715",
              "slightlyInjured": "919",
              "victimClass": "passengers",
            },
            Object {
              "addedon": "2019-04-23",
              "fatalties": "204",
              "fill": "green",
              "id": 5,
              "seriouslyInjured": "356",
              "slightlyInjured": "98",
              "victimClass": "motor_cyclists",
            },
          ]
        }
        key="4"
      >
        <td
          className="font-weight-bold summary_motor_cyclists_2020"
          data-hover="summary_motor_cyclists_2020"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          658
        </td>
      </TableFoot>
    </tr>
  </tfoot>
</table>
`;
